packages/migrate/src/plugins/lint.plugin.ts

```diff
@@ -7,23 +7,24 @@ import { Plugin, PluginParams, PluginResult } from '../interfaces/plugin';
  */
 export default class LintPlugin extends Plugin {
   async run(params: PluginParams<undefined>): PluginResult {
-    const text = this.service.getFileText(params.fileName);
+    const { fileName } = params;
+    const text = this.service.getFileText(fileName);
 
     try {
       const eslint = new ESLint({ fix: true, useEslintrc: true });
-      const [report] = await eslint.lintText(text, { filePath: params.fileName });
+      const [report] = await eslint.lintText(text, { filePath: fileName });
 
       if (report && report.output && report.output !== text) {
         this.logger?.debug(`Plugin 'Lint' run on ${params.fileName}`);
-
-        return report.output;
+        this.service.setFileText(fileName, report.output);
+        return [fileName];
       }
     } catch (e) {
       this.logger?.error(`Plugin 'Lint' failed on ${params.fileName}: ${(e as Error).message}`);
     }
 
     this.logger?.debug(`Plugin 'Lint' run on ${params.fileName} with no changes`);
 
-    return text;
+    return [];
```

Comment:
Confirm we no longer need to return the text

Reply:
If there's changed text, we use setFileText() to commit the text to this file in memory and we only return the filename. If no file has changed, we return an empty array.